WD-Firmware/psp/api_inc/psp_defines.h

```diff
@@ -37,6 +37,144 @@
 
 /* function return codes */
 #define D_PSP_SUCCESS                      0x00
-#define D_PSP_FAIL                          0x01
+#define D_PSP_FAIL                         0x01
 #define D_PSP_PTR_ERROR                    0x02
+
+
+#if __riscv_xlen == 64
+	#define D_pspWORD_SIZE 8
+	#define D_STORE        sd
+	#define D_LOAD         ld
+    #define D_ADDI         addiw
+    #define D_SLL32        sllw
+    #define D_LWU          lwu
+#elif __riscv_xlen == 32
+	#define D_pspWORD_SIZE 4
+	#define D_STORE        sw
+	#define D_LOAD         lw
+    #define D_ADDI         addi
+    #define D_SLL32        sll
+    #define D_LWU          lw
+#else
+	#error Assembler did not define __riscv_xlen
+#endif
+
+
+/* Definitions for the various fields in the CSRs */
+
+#define D_MSTATUS_UIE         0x00000001
+#define D_MSTATUS_SIE         0x00000002
+#define D_MSTATUS_HIE         0x00000004
+#define D_MSTATUS_MIE         0x00000008
+#define D_MSTATUS_UPIE        0x00000010
+#define D_MSTATUS_SPIE        0x00000020
+#define D_MSTATUS_HPIE        0x00000040
+#define D_MSTATUS_MPIE        0x00000080
+#define D_MSTATUS_SPP         0x00000100
+#define D_MSTATUS_HPP         0x00000600
+#define D_MSTATUS_MPP         0x00001800
+#define D_MSTATUS_FS          0x00006000
+#define D_MSTATUS_XS          0x00018000
+#define D_MSTATUS_MPRV        0x00020000
+#define D_MSTATUS_PUM         0x00040000
+#define D_MSTATUS_MXR         0x00080000
+#define D_MSTATUS_VM          0x1F000000
+#define D_MSTATUS32_SD        0x80000000
+#define D_MSTATUS64_SD        0x8000000000000000
+
+#define D_SSTATUS_UIE         0x00000001
+#define D_SSTATUS_SIE         0x00000002
+#define D_SSTATUS_UPIE        0x00000010
+#define D_SSTATUS_SPIE        0x00000020
+#define D_SSTATUS_SPP         0x00000100
+#define D_SSTATUS_FS          0x00006000
+#define D_SSTATUS_XS          0x00018000
+#define D_SSTATUS_PUM         0x00040000
+#define D_SSTATUS32_SD        0x80000000
+#define D_SSTATUS64_SD        0x8000000000000000
+
+#define D_DCSR_XDEBUGVER      (3U<<30)
+#define D_DCSR_NDRESET        (1<<29)
+#define D_DCSR_FULLRESET      (1<<28)
+#define D_DCSR_EBREAKM        (1<<15)
+#define D_DCSR_EBREAKH        (1<<14)
+#define D_DCSR_EBREAKS        (1<<13)
+#define D_DCSR_EBREAKU        (1<<12)
+#define D_DCSR_STOPCYCLE      (1<<10)
+#define D_DCSR_STOPTIME       (1<<9)
+#define D_DCSR_CAUSE          (7<<6)
+#define D_DCSR_DEBUGINT       (1<<5)
+#define D_DCSR_HALT           (1<<3)
+#define D_DCSR_STEP           (1<<2)
+#define D_DCSR_PRV            (3<<0)
+
+#define D_DCSR_CAUSE_NONE     0
```

Comment:
check which CSR is only related to HiFive1 and put them in #if

Reply:
I would like to remove the definition groups that are not in use